tenant/src/main/scala/com/improving/app/tenant/domain/Tenant.scala

```diff
@@ -43,63 +43,73 @@ object Tenant {
     )
   }
 
-  private val commandHandler: (TenantState, TenantCommand) => ReplyEffect[TenantEvent, TenantState] = { (state, command) =>
-    val result: Either[Error, TenantEvent] = state match {
+  private val commandHandler: (TenantState, TenantCommand) => ReplyEffect[TenantResponse, TenantState] = { (state, command) =>
+    val result: Either[Error, TenantResponse] = state match {
       case UninitializedTenant =>
-        command.request.asMessage.sealedValue match {
-          case TenantRequestMessage.SealedValue.EstablishTenantValue(value) => establishTenant(value)
+        command.request match {
+          case x: EstablishTenant => establishTenant(x)
+          case x: GetOrganizations => getOrganizations(x)
           case _ => Left(StateError("Tenant is not established"))
         }
       case establishedState: EstablishedTenantState =>
-        command.request.asMessage.sealedValue match {
-          case TenantRequestMessage.SealedValue.Empty => Left(StateError("Command is not supported"))
-          case TenantRequestMessage.SealedValue.EstablishTenantValue(_) => Left(StateError("Tenant is already established"))
-          case TenantRequestMessage.SealedValue.ActivateTenantValue(value) => activateTenant(establishedState, value)
-          case TenantRequestMessage.SealedValue.SuspendTenantValue(value) => suspendTenant(establishedState, value)
-          case TenantRequestMessage.SealedValue.EditInfoValue(value) => editInfo(establishedState, value)
+        command.request match {
+          case _: EstablishTenant => Left(StateError("Tenant is already established"))
+          case x: ActivateTenant => activateTenant(establishedState, x)
+          case x: SuspendTenant => suspendTenant(establishedState, x)
+          case x: EditInfo => editInfo(establishedState, x)
+          case x: GetOrganizations => getOrganizations(x, Some(establishedState))
+          case _ => Left(StateError("Command is not supported"))
         }
```

Comment:
Why is there no handler for TerminatedTenantState?

Reply:
Not sure. @dag-yoppworks should know why, after his changes to Tenant in the previous PR.

Reply:
Terminated handler does not have to be included here, but if not please create another ticket. It should be small enough that including it here is not too bad, but up to you

Reply:
I'll create another ticket as to limit the scope of this PR. https://yoppworks.atlassian.net/browse/IA-169